datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -1519,18 +1599,85 @@ pub fn ts_scalar_interval_op(
             array.data_type()
         )))?,
     };
-    Ok(ColumnarValue::Array(ret))
+    Ok(ret)
+}
+
+/// This function handles timestamp +/- interval operations where the former is
```

Comment:
Minor suggestion is to move these functions into datafusion/physical-expr/src/expressions/binary/kernels_arrow.rs so they are more discoverable